### PR TITLE
Eric theradia emotion

### PR DESCRIPTION
**Pull request by ericbolo**
(Structure has changed quite a bit w.r.t to task-specific logic/modules)

### CODE REVIEW

File: backend/src/labelit/plugins/emotion_categorical/models/emotion_categorical_task.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def validate_labels(self, labels, is_final):
+        pass  # TODO
+
+    def _get_stats(
+        self,
+        done_annotations,
+    ):
+        stats = {}
+        # TODO
```
Comment (corentingiraud) [inline backend/src/labelit/plugins/emotion_categorical/models/emotion_categorical_task.py:20]: An oversight ?

File: backend/src/labelit/plugins/emotion_categorical/models/emotion_categorical_task.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        done_annotations = Annotation.objects.filter(
+            batch=batch,
+            task=self,
+            is_done=True,
+        )
+
+        return self._get_stats(done_annotations)
+
+    def get_agreement_stats(self, batch):
+        # raise NotImplementedError
```
Comment (corentingiraud) [inline backend/src/labelit/plugins/emotion_categorical/models/emotion_categorical_task.py:43]: Should be removed or implemented ?

File: backend/src/labelit/plugins/realtime_video_dimensional/models/realtime_video_dimensional_label.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from django.db import models
+from labelit.models.label import Label
+
+
+CHOICES = range(0, 101)
```
Comment (corentingiraud) [inline backend/src/labelit/plugins/realtime_video_dimensional/models/realtime_video_dimensional_label.py:5]: Where is it used ?

File: backend/src/labelit/plugins/realtime_video_dimensional/models/realtime_video_dimensional_task.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    def validate_labels(self, labels, is_final):
+        pass  # TODO
+
+    def _get_stats(
+        self,
+        done_annotations,
+    ):
+        stats = {}
+        # TODO
```
Comment (corentingiraud) [inline backend/src/labelit/plugins/realtime_video_dimensional/models/realtime_video_dimensional_task.py:23]: An oversight ?

File: backend/src/labelit/validators/base_realtime_dimensional_annotation_validator.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+            if (
+                c not in valid_chars_set and not c == "\n"
+            ):  # allowing new lines as well as valid characters
+                errors.append((pos, self.case1_error))
+                break
+        if len(errors):
+            raise ValidationError(
+                "; ".join([e[1] + " at character index: " + str(e[0]) for e in errors])
+            )
+        """
```
Comment (corentingiraud) [inline backend/src/labelit/validators/base_realtime_dimensional_annotation_validator.py:20]: The whole file should be removed ?

File: frontend/src/composables/video_player.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+const playerOptions = ref({
+  // videojs options
+  muted: false,
+  language: 'en',
+  controls: true,
+  //playbackRates: [0.7, 1.0, 1.5, 2.0],
+  sources: [
+    {
+      type: 'video/mp4',
+      //src: 'https://cdn.theguardian.tv/webM/2015/07/20/150716YesMen_synd_768k_vp8.webm',
```
Comment (corentingiraud) [inline frontend/src/composables/video_player.js:14]: Should be removed

**Review by corentingiraud — CHANGES_REQUESTED** [review]
Pretty big PR 😄  Can you please try to split your work into smaller pieces in the future ?

- Miss documentation for `VIDEO_FILE_STORAGE` and `AWS_VIDEO_STORAGE_BUCKET_NAME` in `documentation/docs/deployment/03.environment_variables.md`

Very good work otherwise 👍

File: frontend/src/components/LabelitVideoPlayer.vue [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+          setInterval(() => {
+            if (this.player.el && !player_loaded) {
+              this.player.src({
+                type: 'video/mp4',
+                src: res.data.url,
+              });
+              player_loaded = true;
+              resolve();
+            }
+          }, 1000);
```
[the remark this answers is not in the document]
Comment (ericbolo) [reply frontend/src/components/LabelitVideoPlayer.vue:64]: This is  a hack. I will leave you to try it, if you can remove the setInterval, I'm for it

File: backend/src/labelit/plugins/realtime_video_dimensional/models/realtime_video_dimensional_task.py (hunk 4 above)
Comment (ericbolo) [reply backend/src/labelit/plugins/realtime_video_dimensional/models/realtime_video_dimensional_task.py:23]: No, for now this task won't return stats, it's not needed for the core demo of feature so I left it for later

**Comment by corentingiraud** [comment]
I've got this error when I ran the migrations:

```
Module not found !! Plugin: live_correct, dotted path: labelit.plugins.live_correct.serializers.live_correct_task_serializer.CreateOrUpdateLiveCorrectTaskSerializer
Module not found !! Plugin: nested_categorical, dotted path: labelit.plugins.nested_categorical.serializers.nested_categorical_task_serializer.CreateOrUpdateNestedCategoricalTaskSerializer
```

**Comment by corentingiraud** [comment]
You have added some video in the example dataset folder, but not inside the ZIP folder. Is it because the upload feature is not ready yet ?

File: backend/src/labelit/models/document.py [hunk 8]
```diff
@@ -10,6 +10,9 @@ class Document(models.Model):
         blank=True,
         null=True,
     )
+    video_filename = models.CharField(
```
Comment (corentingiraud) [inline backend/src/labelit/models/document.py:13]: Another solution would be to add an enum field called `type` which reference the type of the document. Then rename `audio_filename` by `filename`. This allows to easily add new document type in the future. I don't know what do you think about this ?

**Pull request merged**